### ISSUE TITLE
🚀 Feature: Bring in eslint-plugin-node-dependencies's valid-semver rule

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/eslint-plugin-package-json#49]
Splitting out from https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/43#issuecomment-1727005424 & #42: `eslint-plugin-node-dependencies` has a `valid-semver` rule. Let's merge it into this package!

https://ota-meshi.github.io/eslint-plugin-node-dependencies/rules/valid-semver.html

~Blocked on #40, but once that PR is merged this will be good to go.~ ✅

**Comment by JoshuaKGoldberg** [comment]
Note that this is different from #130 -> #135. `valid-version` is for the root-level version. This is for individual package versions.

**Comment by michaelfaith** [comment]
Are you thinking this would just be incorporated into the valid-package-definition rule / PJV to tighten the testing of the version version check that it does, or as its own rule?  It feels like it could be its own rule.  So the valid-package-definition is more about the shape of the package.json being valid, whereas this would be a more granular check of the version ranges?  What do you think?

**Comment by JoshuaKGoldberg** [comment]
That's what I'm thinking, yeah. I think `valid-package-definition` is an extremely wide rule and it would makes sense to split it up into more targeted rules. https://github.com/JoshuaKGoldberg/package-json-validator/issues/80

**Comment by michaelfaith** [comment]
I see.  So this work would start in PJV making it more modular.  That's kind of adjacent to this comment I made in the TS PR, suggesting that the large single named export could be deprecated in favor of exporting the individual functions and constants directly: https://github.com/JoshuaKGoldberg/package-json-validator/pull/110/files#diff-1df698d75ae20791f18d3bf63174980da33961f49732636648f671b1921929bfR509-R510

So you're seeing the logic for this still living in PJV, but that it's just broken up more so that this plugin could use different validations for different rules?

**Comment by JoshuaKGoldberg** [comment]
Oh TIL I'm not watching all events on that repo. I'd missed that TS PR! Thanks 😄.

And yes, that's what I'm envisioning. I think there will always be a use case for `package-json-validator` to run all checks the way it does now. But the underlying individual functions would be used by this package and potentially others, yes.

**Comment by michaelfaith** [comment]
In light of our discussion in #51, i want to mention that I think this rule is separate and still useful outside of the `valid-<property>` rules. In fact, to avoid mistaken association with the others, i wonder if we should name this `restrict-dependency-versions` or `enforce-dependency-range`?

**Comment by michaelfaith** [comment]
I think this is probably duplicated by our new valid-*Dependencies rules?
